### DISCUSSION TITLE
Keeping the same numbering of headings from docx to html

### DISCUSSION THREAD

**Discussion by didmar** [jgm/pandoc#9674] [Q&A] [answered]
I'm trying to convert docx files into markdown while keeping the headings and their numbering.

Here is an example:
![image](https://github.com/jgm/pandoc/assets/10000597/aacdac7f-dda4-45b9-a631-dd8bbabfc59e)
[example.docx](https://github.com/jgm/pandoc/files/15048205/example.docx)

The output I would like:
```markdown
# LOREM IPSUM (h1)

## 1. Lorem ipsum (h2)

Lorem ipsum 

# LOREM IPSUM (h1)

### 2. Lorem ipsum (h3)

2.1. Lorem ipsum (paragraph)
```

When converting directly to markdown, numbering would disappear, so I'm first converting to html with the `--number-sections ` flag.

Running `pandoc --number-sections -t html example.docx`, I get:
```html
<h1 class="unnumbered" id="lorem-ipsum-h1">LOREM IPSUM (h1)</h1>
<h2 data-number="0.1" id="lorem-ipsum-h2">
   <span class="header-section-number">0.1</span> Lorem ipsum (h2)
</h2>
<p>Lorem ipsum</p>
<h1 class="unnumbered" id="lorem-ipsum-h1-1">LOREM IPSUM (h1)</h1>
<h3 data-number="0.1.1" id="lorem-ipsum-h3">
   <span class="header-section-number">0.1.1</span> Lorem ipsum (h3)
</h3>
<ol type="1">
   <li>
      <p>Lorem ipsum (paragraph)</p>
   </li>
</ol>
```

There are two issues here:
- After an unnumbered heading, the sub-heading gets numbered `0.X`, rather than `X`
- That aside, the numbers are not incremented as in my docx: the h3 should be numbered 2. rather than 0.1.1

My guess is that there is some style information that is not used by Pandoc.

Is there a general solution for that (each of my docx will have different styling)?
Or should I seek an alternative solution such as processing the document XML directly, or exporting to PDF first?
Thank you for your help!

Pandoc version: 3.1.12.3

**Comment by jgm** [comment, marked as answer]
Pandoc just ignores the numbers in Word.  Only the section heading levels are recorded.

**Reply by didmar** [reply]
Thank you for the clarification @jgm!
